controllers/reconciliation/manager.go

```diff
@@ -188,3 +190,30 @@ func (r *ReconcileManager) mergeEnvVars(deployment *v1.Deployment,
 	deployment.Spec.Template.Spec.Containers[0].Env = envVars
 	return deployment
 }
+
+func (r *ReconcileManager) applyNewEnvVar(deployment *v1.Deployment, newEnvs []corev1.EnvVar) *v1.Deployment {
+
+	actualEnvMap := utils.ConvertEnvVarSliceToMap(deployment.Spec.Template.Spec.Containers[0].Env)
+	for _, env := range newEnvs {
+		if _, found := actualEnvMap[env.Name]; !found {
+			deployment.Spec.Template.Spec.Containers[0].Env = append(deployment.Spec.Template.Spec.Containers[0].Env, env)
+		}
+	}
+
+	return deployment
+}
+
+type applicationEnvVar map[string]func(r *ReconcileManager, cr *v1alpha1.EntandoAppV2) string
+
+type listApplicationEnvVar map[string]applicationEnvVar
+
+func (r *ReconcileManager) envVarByVersion(cr *v1alpha1.EntandoAppV2, mapEnv listApplicationEnvVar) []corev1.EnvVar {
+	var newAppEnvs []corev1.EnvVar = make([]corev1.EnvVar, 0)
+	if envs, ok := mapEnv[cr.Spec.Version]; ok {
+		for key, value := range envs {
+			newAppEnvs = append(newAppEnvs, corev1.EnvVar{Name: key, Value: value(r, cr)})
+		}
+		return nil
```

Comment:
Sorry, maybe it's a dumb question, but why returning nil after performing the append?